thirdeye/thirdeye-frontend/app/pods/components/rootcause-chart/component.js

```diff
@@ -115,6 +81,139 @@ export default Ember.Component.extend({
     }
   ),
 
+  series: Ember.computed(
+    'entities',
+    'timeseries',
+    'selectedUrns',
+    'anomalyRange',
+    'baselineRange',
+    'timeseriesMode',
+    function () {
+      const { selectedUrns } = this.getProperties('selectedUrns');
+      return this._makeSeries(selectedUrns);
+    }
+  ),
+
+  //
+  // split view logic
+  //
+
+  splitSeries: Ember.computed(
+    'entities',
+    'timeseries',
+    'selectedUrns',
+    'anomalyRange',
+    'baselineRange',
+    'timeseriesMode',
+    function () {
+      console.log('rootcauseChart: splitSeries()');
+      const { selectedUrns, timeseriesMode } = this.getProperties('selectedUrns', 'timeseriesMode');
+
+      if (timeseriesMode != TIMESERIES_MODE_SPLIT) {
+        return {};
+      }
+
+      console.log('rootcauseChart: splitLabels: generating series');
+      const splitSeries = {};
+      const metricUrns = new Set(filterPrefix(selectedUrns, ['thirdeye:metric:', 'frontend:baseline:metric:']));
+      const otherUrns = new Set([...selectedUrns].filter(urn => !metricUrns.has(urn)));
+
+      filterPrefix(metricUrns, 'thirdeye:metric:').forEach(urn => {
+        const splitMetricUrns = [urn];
+        const baselineUrn = toBaselineUrn(urn);
+        if (metricUrns.has(baselineUrn)) {
+          splitMetricUrns.push(baselineUrn);
+        }
+
+        const splitUrns = new Set(splitMetricUrns.concat([...otherUrns]));
+
+        splitSeries[urn] = this._makeSeries(splitUrns);
+      });
+
+      return splitSeries;
+    }
+  ),
+
+  splitUrns: Ember.computed(
+    'entities',
+    'selectedUrns',
+    'timeseriesMode',
+    function () {
+      console.log('rootcauseChart: splitUrns()');
+      const { entities, selectedUrns, timeseriesMode } =
+        this.getProperties('entities', 'selectedUrns', 'timeseriesMode');
+
+      if (timeseriesMode != TIMESERIES_MODE_SPLIT) {
+        return {};
+      }
+
+      console.log('rootcauseChart: splitLabels: generating urns');
+      return filterPrefix(selectedUrns, 'thirdeye:metric:')
+        .map(urn => [entities[urn].label.split("::")[1], urn])
+        .sort()
+        .map(t => t[1]);
+    }
+  ),
+
+  splitLabels: Ember.computed(
+    'entities',
+    'selectedUrns',
+    'timeseriesMode',
+    function () {
+      console.log('rootcauseChart: splitLabels()');
+      const { entities, selectedUrns, timeseriesMode } =
+        this.getProperties('entities', 'selectedUrns', 'timeseriesMode');
+
+      if (timeseriesMode != TIMESERIES_MODE_SPLIT) {
+        return {};
+      }
+
+      console.log('rootcauseChart: splitLabels: generating labels');
+      return filterPrefix(selectedUrns, 'thirdeye:metric:')
+        .reduce((agg, urn) => { agg[urn] = entities[urn].label.split("::")[1]; return agg; }, {});
```

Comment:
nit: let's break this up into multipe line. `entities[urn].label.split("::")[1]; return agg;`

Reply:
fixed